gui/functionality/idact_notebook.py

```diff
@@ -160,71 +153,92 @@ def deploy_notebook(self):
             sleep_until_allocation_ends(nodes=nodes)
         return
 
-    def open_new_native_argument(self):
-        self.add_argument_window.show()
-    
-    def open_remove_native_argument(self):
-        self.remove_argument_window.show()
-    
-    def open_show_native_argument(self):
-        native_args_list = self.native_args_saver.get_native_args_list()
-        counter = len(native_args_list)
-        self.show_native_arguments_window.ui.table_widget.setRowCount(counter)
-        self.show_native_arguments_window.ui.table_widget.setColumnCount(2)
-
-        for i in range(counter):
-            self.show_native_arguments_window.ui.table_widget.setItem(i, 0, QTableWidgetItem(native_args_list[i][0]))
-            self.show_native_arguments_window.ui.table_widget.setItem(i, 1, QTableWidgetItem(native_args_list[i][1]))
-        self.show_native_arguments_window.show()
-
-    def add_new_native_argument(self):
-        argument_name = self.add_argument_window.ui.argument_name_edit.text()
-        self.parameters['add_native_arguments']['argument_name'] = argument_name
-        value = self.add_argument_window.ui.value_name_edit.text()
-        self.parameters['add_native_arguments']['value'] = value
-        self.saver.save(self.parameters)
+    def get_prepared_native_args(self):
+        args = self.native_args_saver.get_native_args().copy()
+        keys = list(args)
```

Comment:
This line looks a little bit odd to me, I'm not sure but I think the `keys = args.keys()` is better approach.

Reply:
Here is a little bit about it: https://stackoverflow.com/questions/16819222/how-to-return-dictionary-keys-as-a-list-in-python

I think this is ok, but maybe it will be easier to read with your version so I can change it.